src/components/Layout/Navbar/components/UserProfile.tsx

```diff
@@ -1,34 +1,40 @@
-import { useGetSelfDetailsQuery } from "../../../../services/serverApi";
 import { Box, Text } from "@chakra-ui/react";
 import Image from "next/image";
+import { useSelector } from "react-redux";
+import { RootState } from "@/src/store";
 
 import { Dispatch, SetStateAction } from "react";
 
 import styles from "./userProfile.module.css";
 
-export default function UserProfile({ isDropdownVisible, setIsDropdownVisible} : {
+export default function UserProfile({
+  isDropdownVisible,
+  setIsDropdownVisible,
+}: {
   isDropdownVisible: boolean;
   setIsDropdownVisible: Dispatch<SetStateAction<boolean>>;
 }) {
-  const { data: user, isLoading } = useGetSelfDetailsQuery();
-  const imageToShow = user?.picture?.url || '/images/Avatar.png';
+  const { first_name, imageURL } = useSelector(
+    (state: RootState) => state.global
+  );
+
+  const imageToShow = imageURL || "/images/Avatar.png";
 
-  if (isLoading) return <></>;
   return (
-      <Box
-        className={styles.userprofile_container}
-        onClick={() => setIsDropdownVisible(!isDropdownVisible)}
-      >
-        <Text
-          className={styles.userprofile_user__first_name}
-        >{`Hello, ${user?.first_name}`}</Text>
-        <Image
-          src={imageToShow}
-          style={{ borderRadius: '50%' }}
-          width={32}
-          height={32}
-          alt=''
-        />
-      </Box>
+    <Box
+      className={styles.userprofile_container}
+      onClick={() => setIsDropdownVisible(!isDropdownVisible)}
+    >
+      <Text
+        className={styles.userprofile_user__first_name}
+      >{`Hello, ${first_name}`}</Text>
+      <Image
+        src={imageToShow}
+        style={{ borderRadius: "50%" }}
+        width={32}
+        height={32}
+        alt=""
```

Comment:
why is it empty?

Reply:
you can use first name here

Reply:
sure

Reply:
https://github.com/Real-Dev-Squad/members-site/pull/118#discussion_r1407648519